### PR TITLE
test(e2e): test to verify large number of pvc create and delete operation

### PR DESCRIPTION
**Pull request by rohan2794**
JIRA: https://mayadata.atlassian.net/browse/MQ-2632

### CODE REVIEW

File: src/common/e2e_config/e2e_config.go [hunk 1]
```diff
@@ -266,6 +266,12 @@ type E2EConfig struct {
 		FileSystemType    string `yaml:"fileSystemType" env-default:"ext4"`
 		NumberOfOperation int    `yaml:"numberOfOperation" env-default:"9977777"`
 	} `yaml:"fsxExt4Stress"`
+	PvcCreateDelete struct {
+		Replicas         int `yaml:"replicas" env-default:"3"`
+		VolSize          int `yaml:"volMb" env-default:"64"`
+		Iterations       int `yaml:"iterations" env-default:"1"`
+		VolumeMultiplier int `yaml:"volumeMultiplier" env-default:"50"`
```
Comment (blaisedias) [inline src/common/e2e_config/e2e_config.go:273]: Maybe increase volumeMultiplier to 110

File: src/MQ-2632-pvc_create_delete/util.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+func (c *pvcCreateDeleteConfig) pvcCreateDeleteTest() {
+	c.createStorageClass()
+	for ix := 0; ix < c.iterations; ix++ {
+		for _, pvcName := range c.pvcNames {
+			c.createPVC(pvcName)
+		}
+		//Wait for 5 mins. This step makes sure that all the volume creation requests have been sent to csi controller pod
+		logf.Log.Info("Sleep for five minutes")
+		time.Sleep(5 * time.Minute)
```
Comment (blaisedias) [inline src/MQ-2632-pvc_create_delete/util.go:78]: Maybe make this delay configurable.

**Review by blaisedias — APPROVED** [review]
LGTM but see comments

File: src/MQ-2632-pvc_create_delete/util.go (hunk 2 above)
Comment (rohan2794) [reply src/MQ-2632-pvc_create_delete/util.go:78]: good point

File: src/common/e2e_config/e2e_config.go (hunk 1 above)
Comment (rohan2794) [reply src/common/e2e_config/e2e_config.go]: Done

**Review by chriswldenyer — APPROVED** [review]
Some comments, and what looks like some redundant code.

File: src/MQ-2632-pvc_create_delete/config.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+func generatePvcCreateDeleteConfig(testName string, volCount int) *pvcCreateDeleteConfig {
+	params := e2e_config.GetConfig().PvcCreateDelete
+	c := &pvcCreateDeleteConfig{
+		protocol:       common.ShareProtoNvmf,
+		volType:        common.VolFileSystem,
+		fsType:         common.Ext4FsType,
+		volBindingMode: storageV1.VolumeBindingImmediate,
+		pvcSize:        params.VolSize,
```
[the remark this answers is not in the document]
Comment (rohan2794) [reply src/MQ-2632-pvc_create_delete/config.go]: Done

File: src/MQ-2632-pvc_create_delete/config.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	protocol       common.ShareProto
+	fsType         common.FileSystemType
+	volType        common.VolumeType
+	volBindingMode storageV1.VolumeBindingMode
+	replicas       int
+	iterations     int
+	scName         string
+	pvcNames       []string
+	pvcSize        int
+	optsList       []coreV1.PersistentVolumeClaim
```
[the remark this answers is not in the document]
Comment (rohan2794) [reply src/MQ-2632-pvc_create_delete/config.go]: Well spotted, it was due to copy paste from another test. Thanks
Done

**Review by chriswldenyer — APPROVED** [review]
LGTM

**Review by payes — APPROVED** [review]

**Comment by rohan2794** [comment]
bors merge

**Pull request merged**